allennlp/data/dataset_readers/squad.py

```diff
@@ -18,6 +19,158 @@
+        We similarly use this for both the question and the passage.  See :class:`TokenIndexer`.
+    """
+    def __init__(self,
+                 tokenizer: Tokenizer = WordTokenizer(),
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._tokenizer = tokenizer
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    def read(self, file_path: str):
+
+        logger.info("Reading file at %s", file_path)
+        with open(file_path) as dataset_file:
+            dataset_json = json.load(dataset_file)
+            dataset = dataset_json['data']
+        logger.info("Reading the dataset")
+        instances = []
+        for article in tqdm(dataset):
+            for paragraph_json in article['paragraphs']:
+                paragraph = paragraph_json["context"]
+                # replace newlines in the paragraph
+                cleaned_paragraph = paragraph.replace("\n", " ")
+
+                # We add a special token to the end of the passage.  This is because our span
+                # labels are end-exclusive, and we do a softmax over the passage to determine span
+                # end.  So if we want to be able to include the last token of the passage, we need
+                # to have a special symbol at the end.
+                tokenized_paragraph = self._tokenizer.tokenize(cleaned_paragraph) + ['@@STOP@@']
+
+                for question_answer in paragraph_json['qas']:
+                    question_text = question_answer["question"].strip().replace("\n", "")
+                    tokenized_question = self._tokenizer.tokenize(question_text)
+                    print("question_text:", question_text)
+                    print("tokenized:", tokenized_question)
+
+                    # There may be multiple answer annotations, so pick the one that occurs the
+                    # most.
+                    candidate_answers = Counter() # type: Counter
+                    for answer in question_answer["answers"]:
+                        candidate_answers[(answer["answer_start"], answer["text"])] += 1
+                    char_span_start, answer_text = candidate_answers.most_common(1)[0][0]
+
+                    # SQuAD gives answer annotations as a character index into the paragraph, but
+                    # we need a token index for our models.  We convert them here.
+                    char_span_end = char_span_start + len(answer_text)
+                    span_start, span_end = _char_span_to_token_span(paragraph,
+                                                                    tokenized_paragraph,
+                                                                    (char_span_start, char_span_end),
+                                                                    self._tokenizer)
+
+                    # We do a deepcopy here to avoid any weird issues with shared state between
```

Comment:
It's because the paragraph is used in multiple instances.  The question isn't.  I updated the comment to make this more clear.